### DISCUSSION TITLE
How can I update the client from Postgres after custom claims have been updated?

### DISCUSSION THREAD

**Discussion by JakesMD** [supabase/supabase#27027] [Questions] [answered]
I've successfully implemented [role-based access control](https://supabase.com/docs/guides/database/postgres/custom-claims-and-role-based-access-control-rbac?queryGroups=language&language=plpgsql). However, for changes to claims to propagate on the client side, the user currently has to log out and log back in.

I attempted to add my custom claims to the `raw_user_meta_data`. However, updating this from Postgres does not trigger an AuthChangeEvent on the client.

I'm updating the claims on every request using the [db_pre_request](https://github.com/burggraf/postgrest-request-processing/tree/main) function.

Is there a way to update the client from Postgres so that changes to custom claims are immediately reflected without requiring the user to log out and log back in?

**Comment by GaryAustin1** [comment, marked as answer]
There is no way.  You have to refresh the session on the client to get the new jwt.  Signout/signIn is one way, but you can also force a refresh session.
Using a table for your grants would be a way to have RLS immediately see the changes.  You can have both a jwt and a table based method if you also need the info on the client.

**Comment by JakesMD** [comment]
As [GaryAustin1](https://github.com/orgs/supabase/discussions/27027#discussioncomment-9680154) mentioned, this is not possible.

My solution is to refresh the session whenever the app is opened. This ensures that any changes to a user's role will be applied the next time they launch the app, which should be satisfactory for most users.

**Reply by iAbhinav** [reply]
i have a nextjs app. where should i refresh the session, so that

1. User always has latest roles
2. There is least performance impact
